web-nextjs/pages/index.tsx

```diff
@@ -49,102 +51,89 @@ function deserializeRequests(bytes: Uint8Array) {
   return requests;
 }
 
-async function reducer(state: State, action: Action): Promise<State> {
-  let serializer = new bcs.BcsSerializer();
-
-  var bytes;
-  switch (action.kind) {
-    case "response":
-      action.response.serialize(serializer);
-
-      bytes = response(serializer.getBytes());
-
-      break;
-    case "message":
-      action.message.serialize(serializer);
-
-      bytes = message(serializer.getBytes());
-
-      break;
-    default:
-      throw new Error();
-  }
-
-  let requests = deserializeRequests(bytes);
-
-  for (const request of requests) {
-    switch (request.body.constructor) {
-      case types.RequestBodyVariantRender:
-        let bytes = view();
-        let viewDeserializer = new bcs.BcsDeserializer(bytes);
-        let viewModel = types.ViewModel.deserialize(viewDeserializer);
-
-        state = {
-          platform: viewModel.platform,
-          image: viewModel.image,
-          fact: viewModel.fact,
-        };
-
-        break;
-      case types.RequestBodyVariantTime:
-        state = await reducer(state, {
-          kind: "response",
-          response: new types.Response(
-            request.uuid,
-            new types.ResponseBodyVariantTime(new Date().toISOString())
-          ),
-        });
-
-        break;
-      case types.RequestBodyVariantPlatform:
-        state = await reducer(state, {
-          kind: "response",
-          response: new types.Response(
-            request.uuid,
-            new types.ResponseBodyVariantPlatform(
-              new UAParser(navigator.userAgent).getBrowser().name || "Unknown"
-            )
-          ),
-        });
-
-        break;
-      case types.RequestBodyVariantKVRead:
-        break;
-      case types.RequestBodyVariantKVWrite:
-        break;
-      case types.RequestBodyVariantHttp:
-        let url = (request.body as types.RequestBodyVariantHttp).value;
-
-        let resp = await fetch(url);
-        let body = await resp.arrayBuffer();
-        let response_bytes = Array.from(new Uint8Array(body));
-
-        state = await reducer(state, {
-          kind: "response",
-          response: new types.Response(
-            request.uuid,
-            new types.ResponseBodyVariantHttp(response_bytes)
-          ),
-        });
-
-        break;
-      default:
-    }
-  }
-
-  return state;
-}
-
 const Home: NextPage = () => {
-  const [state, dispatch] = useAsyncReducer(reducer, initialState);
+  const [state, setState] = useState(initialState);
+
+  // think it would be great if responses could just be messages with an optional uuid field set.
```

Comment:
I don't believe that's possible. The shell does not know what `Message` to send back in response to a capability request, that's what the continuation UUID provides. Each capability can be used from several places in the app (as demonstrated by the Http capability), and the message callback is call site dependent. That's why the separation exists.